pkg/controller/stacks/stack/stack.go

```diff
@@ -160,29 +188,169 @@ func (h *stackHandler) update(ctx context.Context) (reconcile.Result, error) {
 	return reconcile.Result{}, nil
 }
 
-func (h *stackHandler) createNamespacedRoleBinding(ctx context.Context, owner metav1.OwnerReference) error {
-	cr := &rbac.Role{
+// createPersonaClusterRoles creates admin, edit, and view clusterroles that are
+// namespace+stack+version specific
+func (h *stackHandler) createPersonaClusterRoles(ctx context.Context, labels map[string]string) error {
+	// TODO(displague) refine 'editor' persona roles
+	for persona := range roleVerbs {
+		labelsCopy := map[string]string{}
+		for k, v := range labels {
+			labelsCopy[k] = v
+		}
+
+		rules := []rbacv1.PolicyRule{}
+		for _, crd := range h.ext.Spec.CRDs {
+			rules = append(rules, rbacv1.PolicyRule{
+				APIGroups: []string{crd.GroupVersionKind().Group},
+				Resources: []string{crd.Kind},
+				Verbs:     roleVerbs[persona],
+			})
+		}
+		name := fmtPersonaRole(h.ext, persona)
+
+		var crossplaneScope string
+
+		if h.isNamespaced() {
+			crossplaneScope = "namespace"
+		} else {
+			crossplaneScope = "environment"
+		}
+
+		aggregationLabel := fmt.Sprintf(labelAggregateFmt, crossplaneScope, persona)
+
+		cr := &rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   name,
+				Labels: labelsCopy,
+			},
+			Rules: rules,
+		}
+
+		meta.AddLabels(cr, map[string]string{
+			aggregationLabel: "true",
+		})
+
+		meta.AddLabels(cr, stacks.ParentLabels(h.ext))
+
+		if h.isNamespaced() {
+			meta.AddLabels(cr, map[string]string{
+				fmt.Sprintf(labelNamespaceFmt, h.ext.GetNamespace()): "true",
+			})
+		}
+
+		if err := h.kube.Create(ctx, cr); err != nil && !kerrors.IsAlreadyExists(err) {
+			return errors.Wrap(err, "failed to create persona cluster roles")
+		}
+	}
+	return nil
+}
+
+func generateNamespaceClusterRoles(stack *v1alpha1.Stack) (roles []*rbacv1.ClusterRole) {
+	personas := []string{"admin", "edit", "view"}
+
+	namespaced := (stack.Spec.PermissionScope == string(apiextensions.NamespaceScoped))
```

Comment:
there's a new function in this PR that does this same check, right? can we use that? `isNamespaced`

Reply:
Indeed there is.. :point_left: :nerd_face: :airplane: